### DISCUSSION TITLE
issue with career

### DISCUSSION THREAD

**Discussion by Kirtishukla2004** [community/community#29851] [New to GitHub]
hello,
namestey  everyone thanks for checking my issue 
i am kirti a new coder in engineers feild ,as i am new that's why i am so confused no doubt i wanna  make my career in coding but i am too confused that 
that what to do next some basic and stupid quetions are making my mind confuse can you please answer them so that i can take a good step 

- 1)is to learn many computer languages are important ?if yes why:no then which  lanuages which will help me in a best way
- 2)my background is important to take a good job at a respectful company?

**please answer it**

**Comment by radar-blip** [comment]
It might confuse you even more trying to learn many languages all at once. Instead just focus on 1 language. 
That language, ideally, would be one that you might use at your job/work.
Python is a good choice. 😄

**Reply by Kirtishukla2004** [reply]
thanks for answering my issue

**Reply by radar-blip** [reply]
You're welcome. Good luck!

**Comment by jfullstackdev** [comment]
> * 1)is to learn many computer languages are important ?if yes why:no then which  lanuages which will help me in a best way

that will depend on the company tech stack, but it's worth learning those extensively being used in the field



> * 2)my background is important to take a good job at a respectful company?

it's preferred to have IT / CS grad but other companies will really look into your experience, even from the Open Source

**Reply by Kirtishukla2004** [reply]
thanks for checking and aswering my issue

**Comment by opuu** [comment]
## How many languages should I learn?
If you know the core concept of one language, you can jump to any language very easily. So I will recommend learn the core concepts and syntax of one language first. Once you become confident in this language you can start exploring other languages.

I won't recommend starting with python because it hides many important things from you. Start with a curly brace syntax language, like C++ or Java where you have make your hands dirty to do simple tasks. This will teach you many core staffs of programming.

## Is your background/educational qualifications important to get a good job?
Yes, it is important but not a must thing. Having a CS or related degree helps as most companies requires you to have at least undergraduate degree to apply for jobs. But the most important thing is your skill and experience. You can get a good job without academic qualifications. There are more programmers working professionally at good companies without a relevant degree than you think.

**Reply by airtower-luna** [reply]
> I won't recommend starting with python because it hides many important things from you.

What kind of "important things" do you mean? Python provides access to anything above OS level, but a beginner doesn't *have to* worry about a lot of stuff if they don't want to (yet). I consider that a good thing.

Java provides similar abstractions, but you still have to learn to compile things before you can do anything. It's a decent choice (and it was my first programming language, well over 20 years ago), but I wouldn't *specifically* recommend it for beginners.

C++ (or C, or any other language with unchecked memory access) is definitely *not* a good choice as first programming language nowadays that memory-safe alternatives are available. For starting with a systems programming language, Rust or Go are good choices. I particularly like Rust, it lets you do all the low level things and ensures memory safety at compile time.

However there's always one consideration that's more important than any language feature (or lack of feature): Motivation. If someone wants to do something specific and it requires a certain language, that language is the best choice. :slightly_smiling_face:

**Reply by opuu** [reply]
>What kind of "important things" do you mean? Python provides access to anything above OS level, but a beginner doesn't have to worry about a lot of stuff if they don't want to (yet). I consider that a good thing.

Python has a very different syntax than most other languages (For me it looks like pseudocode), so you have to learn a completely new syntax if you want to switch to another language. It is hard for beginners to jump to different languages from python. Also what I believe is, knowing how to manage memory and other low level staffs from the beginning does not hurt you instead makes you a better programmer.

>However there's always one consideration that's more important than any language feature (or lack of feature): Motivation.

Exactly, if you truly want to learn programming, it does not matter which language you use. Motivation and your aim matters.
